repos/system_upgrade/el7toel8/actors/checkchrony/actor.py

```diff
@@ -0,0 +1,53 @@
+from leapp.actors import Actor
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.reporting import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common.reporting import report_generic
+from leapp.libraries.stdlib import CalledProcessError, run
+
+
+class CheckChrony(Actor):
+    """
+    Check for incompatible changes in chrony configuration.
+
+    Warn that the default chrony configuration in RHEL8 uses the leapsectz
+    directive.
+    """
+
+    name = 'check_chrony'
+    consumes = (InstalledRedHatSignedRPM,)
+    produces = (Report)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        installed_packages = set()
+
+        signed_rpms = self.consume(InstalledRedHatSignedRPM)
+        for rpm_pkgs in signed_rpms:
+            for pkg in rpm_pkgs.items:
+                installed_packages.add(pkg.name)
+
+        if 'chrony' not in installed_packages:
+            report_generic(
+                    title='chrony package is not installed',
+                    summary='chrony package was not found in the rpm database',
+                    severity='low')
+            return
+
+        # Check if using default chrony.conf
+        try:
+            run(['/bin/sh', '-c', 'rpm -V chrony | grep chrony.conf'])
+            default_chrony_conf = False
+        except (OSError, CalledProcessError):
+            default_chrony_conf = True
+
+        if default_chrony_conf:
+            report_generic(
+                    title='chrony using default configuration file',
+                    summary='default chrony configuration in RHEL8 uses leapsectz directive, which cannot be used with leap smearing NTP servers, and uses a single pool directive instead of four server directives',
```

Comment:
s/leap/leapp

Reply:
@fernflower nope, this refers to leap as in leap seconds ;-)

Reply:
Oh my, I'll go brew some more coffee, thanks :coffee: